objcgen/embedder.cs

```diff
@@ -629,7 +656,8 @@ public int Compile ()
 								mmp.Append ("--debug ");
 							mmp.Append ("-p "); // generate a plist
 							mmp.Append ($"--target-framework {GetTargetFramework ()} ");
-							mmp.Append ($"\"--link_flags=-force_load {Path.GetFullPath (sdk_output_file)}\" ");
+							string extensionFlag = Extension ? "-fapplication-extension" : "";
+							mmp.Append ($"\"--link_flags={extensionFlag} -force_load {Path.GetFullPath (sdk_output_file)}\" ");
```

Comment:
@chamons there's an issue here, not related to the PR - it existed before

The whole thing is quoted but if `Path.GetFullPath (sdk_output_file)` returns a path with a space then this will fail.